src/Middleware.php

```diff
@@ -46,6 +46,36 @@ public static function addLogging(HandlerStack $stack, LoggerInterface $logger)
     }
 
     /**
+     * Read Maxemail's deprecation notices from the HTTP Warning header
+     * to log as warning and trigger deprecation notice
+     *
+     * @param HandlerStack $stack
+     * @param LoggerInterface $logger
+     */
+    public static function addWarningLogging(HandlerStack $stack, LoggerInterface $logger)
+    {
+        $middleware = GuzzleMiddleware::mapResponse(function (ResponseInterface $response) use ($stack, $logger) {
```

Comment:
Does this need the `$stack` to be in the closure scope? I can't see it being used...